src/Silex/CallbackResolver.php

```diff
@@ -33,40 +33,59 @@ public function __construct(Container $app)
      */
     public function isValid($name)
     {
-        return is_string($name) && preg_match(static::SERVICE_PATTERN, $name);
+        return is_string($name) && (preg_match(static::SERVICE_PATTERN, $name) || isset($this->app[$name]));
     }
 
     /**
      * Returns a callable given its string representation.
      *
      * @param string $name
      *
-     * @return array A callable array
+     * @return callable A callable value
      *
      * @throws \InvalidArgumentException In case the method does not exist.
      */
     public function convertCallback($name)
     {
-        list($service, $method) = explode(':', $name, 2);
+        if (preg_match(static::SERVICE_PATTERN, $name)) {
+            list($service, $method) = explode(':', $name, 2);
 
-        if (!isset($this->app[$service])) {
-            throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
-        }
+            if (!isset($this->app[$service])) {
+                $this->throwServiceNotExists($service);
```

Comment:
is it really useful to have this? Pimple already throw an exception if the key does not exists

alternatively, the whole if else could be wrapped in a `try catch(\InvalidArgumentException)` and rethrow another with a more explicit message if needed

```php
    public function convertCallback($name)
    {
        try {
            if (preg_match(static::SERVICE_PATTERN, $name)) {
                list($service, $method) = explode(':', $name, 2);
                return array($this->app[$service], $method);
            } else {
                $callback = $this->app[$name];

                if (!is_callable($callback)) {
                    throw new \InvalidArgumentException(sprintf('Service "%s" is not callable.', $name));
                }

                return $callback;
            }
    }
    catch (\InvalidArgumentException $e) {
        throw new \InvalidArgumentException(sprintf('Service "%s" is not a valid callback.', $service), 0, $e);
    }
```